code/game/machinery/computer/security.dm

```diff
@@ -1,46 +1,321 @@
+	var/datum/data/record/active_record_security
+
+/obj/structure/machinery/computer/secure_data/attackby(obj/item/held_object as obj, user as mob)
+	if(istype(held_object, /obj/item/device/clue_scanner) && !scanner)
+		var/obj/item/device/clue_scanner/held_scanner = held_object
+		if(!held_scanner.print_list)
+			to_chat(user, SPAN_WARNING("There are no prints stored in \the [held_scanner]!"))
 			return
 
 		if(usr.drop_held_item())
-			O.forceMove(src)
-			scanner = O
-			to_chat(user, "You insert [O].")
+			held_scanner.forceMove(src)
+			scanner = held_scanner
+			to_chat(user, "You insert [held_scanner].")
+			update_static_data_for_all_viewers()
 
 	..()
 
 /obj/structure/machinery/computer/secure_data/attack_remote(mob/user as mob)
 	return attack_hand(user)
 
-//Someone needs to break down the dat += into chunks instead of long ass lines.
+/obj/structure/machinery/computer/secure_data/ui_data(mob/user)
+	var/list/data = list()
+
+	if(isnull(GLOB.data_core.general) || isnull(GLOB.data_core.security))
+		return
+
+	data["active_record"] = list()
+
+	// Get advanced data only for selected record.
+	if (active_record_general)
+		data["active_record"]["name"] = active_record_general.fields["name"]
+		data["active_record"]["ref"] = "\ref[active_record_general]"
+		data["active_record"]["id"] = active_record_general.fields["id"]
+		data["active_record"]["rank"] = active_record_general.fields["rank"]
+		data["active_record"]["squad"] = active_record_general.fields["squad"]
+		data["active_record"]["sex"] = active_record_general.fields["sex"]
+		data["active_record"]["age"] = active_record_general.fields["age"]
+		data["active_record"]["physical_status"] = active_record_general.fields["p_stat"]
+		data["active_record"]["mental_status"] = active_record_general.fields["m_stat"]
+
+	// Get security data, if it exists.
+	if (active_record_security)
+		data["active_record"]["criminal_status"] = active_record_security.fields["criminal"]
+		data["active_record"]["incidents"] = active_record_security.fields["incidents"]
+		data["active_record"]["comments"] = active_record_security.fields["comments"]
+
+	return data
+
+/obj/structure/machinery/computer/secure_data/ui_static_data(mob/user)
+	var/list/data = list()
+
+	if(isnull(GLOB.data_core.general) || isnull(GLOB.data_core.security))
+		return
+
+	data["wanted_statuses"] = WANTED_STATUSES
+	data["records"] = list()
+
+	// Get basic data from every record.
+	for(var/datum/data/record/record_general_data as anything in GLOB.data_core.general)
+		var/crew_member = list()
+
+		crew_member["name"] = record_general_data.fields["name"]
+		crew_member["ref"] = "\ref[record_general_data]"
+		crew_member["rank"] = record_general_data.fields["rank"]
+
+		for(var/datum/data/record/record_security_data as anything in GLOB.data_core.security)
+			if (record_security_data.fields["id"] == record_general_data.fields["id"])
+				crew_member["criminal_status"] = record_security_data.fields["criminal"]
```

Comment:
are we ok with criminal status not being dynamically updated? With this in static data we would only see data refreshed if the UI was closed and reopened or hard refreshed

Reply:
To get criminal status of every player, you have to search through the security records datacore for EVERY player to locate their record.
For 150 marine players that's just under and very roughly 22,500 checks (150 x 150), every UI update.
Doing this every tick seems like a disgusting waste of processing power for something so unimportant.

Reply:
That makes sense. One possibility would be to have a service go manage who is a criminal - which would reduce the scope but honestly that is way out of scope here